wordpress-mobile/react-native-aztec
# Initial iOS support.

### Description

This PR adds initial support for iOS.  It allows running the project in iOS and loading a very simple (solid blue) view - as a placeholder for Aztec.

### Details:

There are several changes included in this PR due to several issues I had to deal with in iOS:

- Added two identical .gitignore files for Xcode / Swift / ObjC ([one](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-d7c35f6840db1950f2da9713671b70ddR1) for the example project, and [one](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-88fb5bc44eff8c82891374d0ae90f56aR1) for the root project).  They're both a copy of [GitHub's Swift `.gitignore` file](https://github.com/github/gitignore/blob/master/Swift.gitignore).  The reason I kept them in the Xcode project folder is to make it easier to update them by just overwriting the files.  The added bonus is that Android devs don't need to care about iOS's files.  The downside is the `.gitignore` file is duplicated.  Feel free to criticize!
- The root `package.json` [no longer tries to clean the example App](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L8).  The rationale is that the library shouldn't care about the example App's caches.
- Added [a clean and clean:install procedure](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-012ac553d93d987b8d34ad013052bc96R11) to the example app's `package.json` file.
- Added [some iOS cleaning procedures](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-012ac553d93d987b8d34ad013052bc96R12) which shouln't run if Xcode isn't installed.
- Added a `yarn clean cache --force` call to both [the root](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R8) and [the example app's](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-012ac553d93d987b8d34ad013052bc96R11) clean procedures.
- Updated [the `react` and `react-native` dependencies](https://github.com/wordpress-mobile/react-native-aztec/pull/12/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R12) to be compatible with Xcode 9.3.
- Updated `yarn.lock`, although I'm not sure if that's necessary - let me know if it's appropriate to roll that back.

### Additional Info

<details><summary>Expand to see .babelrc error log</summary>

```
error: bundling failed: Error: Unable to resolve module `./node_modules` from `/Users/diego/Developer/react-native-aztec/example/App.js`: The module `./node_modules` could not be found from `/Users/diego/Developer/react-native-aztec/example/App.js`. Indeed, none of these files exist:

  * `/Users/diego/Developer/react-native-aztec/example/node_modules(.native||.ios.js|.native.js|.js|.ios.json|.native.json|.json)`
  * `/Users/diego/Developer/react-native-aztec/example/node_modules/index(.native||.ios.js|.native.js|.js|.ios.json|.native.json|.json)`
    at ModuleResolver.resolveDependency (/Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:161:851)
    at ResolutionRequest.resolveDependency (/Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/node-haste/DependencyGraph/ResolutionRequest.js:91:16)
    at DependencyGraph.resolveDependency (/Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/node-haste/DependencyGraph.js:272:4579)
    at dependencies.map.relativePath (/Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/DeltaBundler/traverseDependencies.js:376:19)
    at Array.map (<anonymous>)
    at resolveDependencies (/Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/DeltaBundler/traverseDependencies.js:374:16)
    at /Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/DeltaBundler/traverseDependencies.js:212:33
    at Generator.next (<anonymous>)
    at step (/Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/DeltaBundler/traverseDependencies.js:297:313)
    at /Users/diego/Developer/react-native-aztec/example/node_modules/metro/src/DeltaBundler/traverseDependencies.js:297:473
 BUNDLE  [ios, dev] ./index.js ░░░░░░░░░░░░░░░░ 0.0% (0/1), failed.
```
</details>

### Important

Please note that I haven't tested how my changes impact Android because I don't have the environment set-up yet (work in progress).  So test carefully and let me know if there's any problem.

### Testing

- Do `yarn:clean install` in the root directory.  Make sure all is fine.
- Do `yarn:clean install` in the `example` directory.  Make sure all is fine and the new iOS cleaning procedures don't show errors.  Make sure `example/node_modules/react-native-aztec` exists.
- Run the Android example project and make sure there are no errors.
- Run the iOS example project and make sure a blue view (placeholder view) is displayed.

<img width="522" alt="screen shot 2018-04-18 at 13 37 02" src="https://user-images.githubusercontent.com/1836005/38962239-c5d1245c-4342-11e8-8865-bfea27d8c375.png">
